File: 1. 단계별로_풀어보기/1~9/9. 약수, 배수와 소수 1단계/1. 5086.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while True:
  A, B = map(int, input().split())

  if A == 0 and B == 0:
    break

  if (A % B) == 0:
    output(f"multiple\n")
  elif (B % A) == 0:
    output(f"factor\n")
  else:
    output(f"neither\n")

File: 1. 단계별로_풀어보기/1~9/9. 약수, 배수와 소수 1단계/1. 5086.js
const fs = require('fs')
const inputArr = fs.readFileSync(0, 'utf-8').toString().trim().split('\n')

inputArr.forEach(inputStr => {
  const [A, B] = inputStr.split(' ').map(Number)

  if (A === 0 && B === 0) return

  let result = ''

  // factor
  if (B % A === 0) {
    result = 'factor'
  } //
  else if (A % B === 0) {
    result = 'multiple'
  } //
  else {
    result = 'neither'
  }
  process.stdout.write(result + '\n')
})
